src/main/java/javaslang/collection/IndexedSeq.java

```diff
@@ -106,7 +106,14 @@ default T last() {
             return get(length() - 1);
         }
     }
-
+    
+    @Override
+    default int lastIndexWhere(Predicate<? super T> p, int end) {
```

Comment:
We prefer readable names for method arguments, e.g. `predicate` instead of `p`. I know that it blows up the code slightly but the code reads better in most cases. However, I think for functions and predicates it is ok to use `f` and `p`. Using `s` for `supplier` might be confusing because `s` is often used for `String`. Just saying in order to talk about some conventions. There is not much documentation on it currently.

We also check objects for `null` in order to give the user good error messages. E.g. here we would do a

```java
Objects.requireNonNull(predicate, "predicate is null");
```

Here is `predicate` better then `p`.